src/Datadog.Trace.ClrProfiler.Managed.Loader/StartupLogger.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Datadog.Trace.ClrProfiler.Managed.Loader
+{
+    internal static class StartupLogger
+    {
+        private const string NixDefaultDirectory = "/var/log/datadog/dotnet";
+
+        private static readonly string LogDirectory = GetLogDirectory();
+        private static readonly string StartupLogFilePath = SetStartupLogFilePath();
+
+        /// <summary>
+        /// Gets a value indicating whether this OS is Windows.
+        /// Prevents the need for a direct System.Runtime reference.
+        /// </summary>
+        public static bool IsWindows
+        {
+            get
+            {
+                var p = (int)Environment.OSVersion.Platform;
+                return (p == 0) || (p == 1) || (p == 2) || (p == 3);
+            }
+        }
+
+        public static void Log(string message, params object[] args)
+        {
+            try
+            {
+                if (StartupLogFilePath != null)
+                {
+                    try
+                    {
+                        using (var fileSink = new FileSink(StartupLogFilePath))
+                        {
+                            fileSink.Info($"[{DateTime.UtcNow}] {message}{Environment.NewLine}", args);
+                        }
+
+                        return;
+                    }
+                    catch
+                    {
+                        // ignore
+                    }
+                }
+
+                Console.Error.WriteLine(message, args);
+            }
+            catch
+            {
+                // ignore
+            }
+        }
+
+        public static void Log(Exception ex, string message, params object[] args)
+        {
+            message = $"{message}{Environment.NewLine}{ex}";
+            Log(message, args);
+        }
+
+        private static string GetLogDirectory()
+        {
+            string logDirectory = null;
+
+            try
+            {
+                var nativeLogFile = Environment.GetEnvironmentVariable("DD_TRACE_LOG_PATH");
+
+                if (!string.IsNullOrEmpty(nativeLogFile))
+                {
+                    logDirectory = Path.GetDirectoryName(nativeLogFile);
+                }
+
+                if (logDirectory == null)
+                {
+                    if (IsWindows)
```

Comment:
I don't think we need the extra (non-inlined) property call.
```suggestion
                    if (Environment.OSVersion.Platform == PlatformID.Win32NT)
```